java/src/test/java/org/rocksdb/SequenceNumberAllocationTest.java

```diff
@@ -0,0 +1,198 @@
+        AtomicBoolean cancellable = new AtomicBoolean(false);
+        Options options = new Options().setCreateIfMissing(true);
+
+        Future<?> task = executor.submit(() -> {
+            try {
+                RocksDB db = RocksDB.open(options, dbFolder.getRoot().getAbsolutePath());
+                int count = fillDatabase(db, writeOptions, 0, 10);
+                assertEquals("Expected 10 values in the database", 10, count);
+                db.flush(new FlushOptions());
+                sequenceNumberAfterFlush.set(db.getLatestSequenceNumber());
+                assertEquals("Sequence number after flush should be 10", 10, sequenceNumberAfterFlush.get());
+                count = fillDatabase(db, writeOptions, 10, 20);
+                assertEquals("Expected 20 values in the database", 20, count);
+                assertEquals("Latest DB sequence number should be 20", 20, db.getLatestSequenceNumber());
+                cancellable.set(true);
+                while(true) {
+                    // Wait indefinitely
+                }
+            }
+            catch (RocksDBException e) {
+                fail("Encountered exception:" + e);
+            }
+        });
+
+        // Enough time to finish performing database operations
+        sleep(5 * 1000);
+
+        if(cancellable.get()) {
+            task.cancel(true);
+            assertTrue("Failed to cancel task", task.isCancelled());
+        } else {
+            fail("Task is not cancellable");
+        }
+
+        executor.shutdownNow();
+        assertTrue("Executor did not shutdown", executor.isShutdown());
+
+        int expectedSequenceNumber = writeOptions.disableWAL() ? 10 : 20;
+        try (RocksDB db = RocksDB.openReadOnly(options, dbFolder.getRoot().getAbsolutePath())) {
+            assertEquals("Lastest sequence number is not the one expected", expectedSequenceNumber, db.getLatestSequenceNumber());
+        } catch (RocksDBException e) {
+            fail("Could not reopen the database:" + e);
+        }
+    }
+
+    @Parameters(name = "{0}")
+    public static Iterable<Object> writeOptionsProvider() {
+        return Arrays.asList(
+                new WriteOptions().setDisableWAL(false),
+                new WriteOptions().setDisableWAL(true)
+        );
+    }
+
+    private static int fillDatabase(RocksDB db, WriteOptions writeOptions, int start, int size)
+            throws RocksDBException
+    {
+        assertEquals("Database does not have expected sequence number " + start, start, db.getLatestSequenceNumber());
+        int count = start;
+        for (int i = start; i < size; ++i) {
+            byte[] key = ("key" + i).getBytes(StandardCharsets.UTF_8);
+            byte[] value = ("value" + i).getBytes(StandardCharsets.UTF_8);
+            db.put(writeOptions, key, value);
+            count++;
+            assertEquals("Sequence number is not updated", count, db.getLatestSequenceNumber());
+        }
+
+        return count;
```

Comment:
is this perhaps just `size - start` or similar?

Reply:
Yes, but I want also the intermediate assertion on line 193.